pkg/controller/installation/windows_controller.go

```diff
@@ -0,0 +1,597 @@
+			return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
+		}
+		if amazonCRDRequired {
+			logw.Info("Rebooting to enable AWS controllers")
+			os.Exit(0)
+		}
+	}
+
+	certificateManager, err := certificatemanager.Create(r.client, &instance.Spec, r.clusterDomain, common.OperatorNamespace(), certificatemanager.AllowCACreation())
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create the Tigera CA", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	typhaNodeTLS, err := GetOrCreateTyphaNodeTLSConfig(r.client, certificateManager)
+	if err != nil {
+		logw.Error(err, "Error with Typha/Felix secrets")
+		r.status.SetDegraded(operatorv1.CertificateError, "Error with Typha/Felix secrets", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	err = utils.GetK8sServiceEndPoint(r.client)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error reading services endpoint configmap", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// Fetch default FelixConfiguration
+	felixConfiguration := &crdv1.FelixConfiguration{}
+	err = r.client.Get(ctx, types.NamespacedName{Name: "default"}, felixConfiguration)
+	if err != nil && !apierrors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Unable to read FelixConfiguration", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	var aci *operatorv1.AmazonCloudIntegration
+	if r.amazonCRDExists {
+		aci, err = utils.GetAmazonCloudIntegration(ctx, r.client)
+		if apierrors.IsNotFound(err) {
+			aci = nil
+		} else if err != nil {
+			r.status.SetDegraded(operator.ResourceReadError, "Error reading AmazonCloudIntegration", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// nodeReporterMetricsPort is a port used in Enterprise to host internal metrics.
+	// Operator is responsible for creating a service which maps to that port.
+	// Here, we'll check the default felixconfiguration to see if the user is specifying
+	// a non-default port, and use that value if they are.
+	nodeReporterMetricsPort := defaultNodeReporterPort
+	var nodePrometheusTLS certificatemanagement.KeyPairInterface
+	if instance.Spec.Variant == operatorv1.TigeraSecureEnterprise {
+
+		// Determine the port to use for nodeReporter metrics.
+		if felixConfiguration.Spec.PrometheusReporterPort != nil {
+			nodeReporterMetricsPort = *felixConfiguration.Spec.PrometheusReporterPort
+		}
+
+		if nodeReporterMetricsPort == 0 {
+			err := errors.New("felixConfiguration prometheusReporterPort=0 not supported")
+			r.status.SetDegraded(operatorv1.InvalidConfigurationError, "invalid metrics port", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+
+		nodePrometheusTLS, err = certificateManager.GetOrCreateKeyPair(r.client, render.NodePrometheusTLSServerSecret, common.OperatorNamespace(), dns.GetServiceDNSNames(render.CalicoNodeMetricsService, common.CalicoNamespace, r.clusterDomain))
```

Comment:
This should be `GetKeyPair` instead of `GetOrCreate` - the core controller should create this for us (and we should add a comment saying as much)